contracts/DDNS.sol

```diff
@@ -0,0 +1,105 @@
+pragma solidity ^0.4.23;
+
+import './abstractions/DDNSTemplate.sol';
+import './common/Destructible.sol';
+import './libs/DomainLib.sol';
+
+contract DDNS is DDNSTemplate, Destructible {
+    using DomainLib for DomainLib.DomainInfo;
+
+    uint public constant REGISTRATION_FEE = 1 ether;
+    uint public constant MIN_DOMAIN_NAME_LENGTH = 5;
+
+    event LogReceipt(address owner, bytes domainName, uint payment, uint expiration);
+    event LogRegisteredNewDomain(bytes domain, bytes4 ip, address owner);
+    event LogTransferOwnership(bytes domain, address newOwner);
+    event LogDomainEdit(bytes domain, bytes4 ip);
+    event LogExtension(bytes domain);
+    event LogWithdrawal();    
+
+    mapping (address=>Receipt[]) public receipts;
+    mapping (bytes=>DomainLib.DomainInfo) domains;
+
+    modifier isValidOwnedDomain(bytes _domain) {
+        require(domains[_domain].exists());
+        require(domains[_domain].getOwner() == msg.sender);
+        require(!domains[_domain].hasExpired());
+        _;
+    }
+
+    // register function is used to both extend owned domain expiration date or purchase a new domain
+    function register(bytes _domain, bytes4 _ip) public payable {
+        require(_domain.length > MIN_DOMAIN_NAME_LENGTH);
+        require(_ip != 0);
+
+        DomainLib.DomainInfo storage domain = domains[_domain];
+
+        uint price;
+        bool isExtending = false;
+
+        if (domain.exists() && !domain.hasExpired()) {
+            require(msg.sender == domain.getOwner());
+            price = REGISTRATION_FEE;
```

Comment:
So if we extend the expiry date of a short-named domain we will pay only the regular fee, but not the bigger one we paid, when we registered the short domain at first?

Reply:
Yes, that is how I understood the extension procedure. If you just want to keep your domain you pay one more ETH every year. Only the first time it cost the actual amount for the domain. Is my assumption wrong? Why should we pay the same price again when we first bought the domain just to keep the domain? Maybe my business logic is flawed :D